File: Level 1/유연근무제/solution.py
def time_to_minute(time):
    hour = time // 100
    minute = time % 100
    return hour * 60 + minute


def solution(schedules, timelogs, startday):
    answer = 0

    for schedule, timelog in zip(schedules, timelogs):
        day = startday
        is_ok = True

        for time in timelog:
            if day <= 5 and time_to_minute(schedule) + 10 < time_to_minute(time):
                is_ok = False
                break
            day = day + 1 if day < 7 else 1

        answer += 1 if is_ok else 0

    return answer

File: Level 1/유연근무제/solution.js
function solution(schedules, timelogs, startday) {
  let answer = 0;
  const n = schedules.length;

  for (let i = 0; i < n; i++) {
    let day = startday;
    let isOk = true;
    const schedule = schedules[i];
    const timelog = timelogs[i];

    for (const time of timelog) {
      if (day <= 5 && timeToMinute(schedule + 10) < timeToMinute(time)) {
        isOk = false;
        break;
      }
      day = day < 7 ? day + 1 : 1;
    }

    if (isOk) answer++;
  }

  return answer;
}

function timeToMinute(time) {
  const hour = Math.floor(time / 100);
  const minute = time % 100;
  return hour * 60 + minute;
}
